examples/hotspot.rs

```diff
@@ -8,22 +8,78 @@ use network_manager::device;
 use network_manager::connection;
 
 
-fn main() {
+struct Options {
+    interface: Option<String>,
+    ssid: String,
+    password: Option<String>,
+}
+
+
+fn print_usage_and_exit() {
+    println!("USAGE: hotspot [-i INTERFACE] SSID [PASSWORD]");
+    process::exit(1);
+}
+
+
+fn parse_options() -> Options {
     let args: Vec<String> = env::args().collect();
 
-    if args.len() < 3 {
-        println!("USAGE: hotspot SSID PASSWORD");
-        process::exit(1);
+    if args.len() < 2 {
+        print_usage_and_exit();
+    }
+
+    let (ssid_pos, interface) = if args[1] == "-i" {
```

Comment:
Do you think its worth replacing all of this with something like https://github.com/kbknapp/clap-rs ?

Or do you wanna do that in a seperate PR afterwards?